File: backend/issues-tracker/shared/src/main/java/com/mateuszziomek/issuestracker/shared/ui/http/rest/v1/dto/user/RegisterUserDto.java
package com.mateuszziomek.issuestracker.shared.ui.http.rest.v1.dto.user;

public record RegisterUserDto(String email, String password) {
    public static final String USER_EMAIL_FIELD_NAME = "email";
    public static final String USER_PLAIN_PASSWORD_FIELD_NAME = "password";
}

File: frontend/src/users/dtos/register-user.dto.ts
import { ApplicationErrorCode } from '../../shared/dtos/application-error.dto';

export interface RegisterFormFields {
  email: string;
  password: string;
  repeatPassword: string;
}

export interface RegisterUserDto {
  email: string;
  password: string;
}

export interface ResponseDataDto {
  code: ApplicationErrorCode;
  status: number;
  message: string;
  details?: {
    someField: string[][];
    anotherField: string[][];
  };
}
